### ISSUE TITLE
Support for ES6 iterators

### ISSUE THREAD

**Issue by segevfiner** [toss/es-toolkit#194]
lodash was built before iterators, most of the operations can achieve better performance/memory usage if they were built around iterators instead, it would be nice if this library supported iterators rather than just plain Arrays/Objects. Taking it a step further than just another lodash.

**Comment by raon0211** [comment]
You're right. We might implement some features implemented in [iterator-polyfill](https://github.com/alkihis/iterator-polyfill) or other libraries.

**Comment by raon0211** [comment]
`es-toolkit/iterator` has been added in [f36f965](https://github.com/toss/es-toolkit/commit/f36f9653) ([#1815](https://github.com/toss/es-toolkit/pull/1815)) and will be included in the next release.

It provides lazy, iterator-based functions built on native Iterator Helper semantics — including a data-last `es-toolkit/fp/iterator` entry point for pipe-friendly composition.

For details, see the documentation: https://github.com/toss/es-toolkit/blob/main/docs/iterator/intro.md (it will be published at https://es-toolkit.dev/iterator/intro.html once the release is out).

Thanks for the suggestion!

**Comment by raon0211** [comment]
Good news — [#1815](https://github.com/toss/es-toolkit/pull/1815) introduced the `es-toolkit/iterator` module: lazy, iterator-based counterparts of the toolkit's utilities (`map`, `filter`, `take`, `drop`, `chunk`, `zip`, `uniqBy`, `scan`, `count`, and more), plus data-last variants in `es-toolkit/fp/iterator` that compose with `pipe`:

```ts
import { pipe } from 'es-toolkit/fp';
import { map, filter, take, toArray } from 'es-toolkit/fp/iterator';

pipe(
  someInfiniteIterable,
  filter(x => x % 2 === 0),
  map(x => x * 10),
  take(5),
  toArray()
);
```

This was Phase 1 and more utilities will follow, but it covers the core of what this issue asked for, so I'm closing it in favor of tracking further additions separately. It ships in the next release. Thanks for the suggestion — and for the patience, this was one of our oldest open issues!
